e2e/saas.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { test, type Page } from '@playwright/test';
+
+test.describe('Sass Challenge', () => {
+  let page: Page;
+
+  test.beforeAll(async ({ browser }) => {
+    page = await browser.newPage();
+    await page.goto(
+      '/learn/front-end-development-libraries/sass/use-for-to-create-a-sass-loop'
+    );
+  });
+
+  test.afterAll(async () => {
+    await page.close();
+  });
+
+  test('should render the sass preview', async () => {
+    const frame = await Promise.resolve(page.mainFrame());
```

Comment:
In this line we are using Promise.resolve() to avoid the TS error `Unexpected 'await' of a non-Promise (non-"Thenable") value`. and page.mainFrame() gets the iframe, here is the doc reference for that [https://playwright.dev/docs/api/class-frame#:~:text=%3B%0A%20%20dumpFrameTree(-,page.mainFrame(),-%2C%20%27%27)](url)